app/main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+import os
+
+
+def copy_file(command: str) -> None:
+    parts = command.split()
+
+    if len(parts) != 3 or parts[0] != "cp":
+        raise ValueError(
+            "Invalid command format. Use 'cp source_file destination_file'."
+        )
+
+    source_file_path = parts[1]
+    destination_file_path = parts[2]
+
+    if source_file_path == destination_file_path:
+        return
+
+    if not os.path.isfile(source_file_path):
+        print(
+            f"Error: Source file '{source_file_path}' "
+            "does not exist or is not a valid file."
+        )
+
+    try:
+        with open(source_file_path, "r") as source_file:
+            with open(destination_file_path, "w") as destination_file:
```

Comment:
Consider using a single 'with' statement to open both files simultaneously, as suggested in the task description. This can be done by separating the context managers with a comma.